purview/api/v1.py

```diff
@@ -79,18 +82,34 @@ def get_resource_story(resource, uid):
     if not item:
         raise NotFound('This item does not exist')
 
-    for _, prop_def in item.__all_properties__:
-        if isinstance(prop_def, UniqueIdProperty):
-            forward_query = create_query(item, prop_def.name, uid)
-            backward_query = create_query(item, prop_def.name, uid, reverse=True)
-            break
+    uid_name = item.unique_id_property
+    forward_query = create_story_query(item, uid_name, uid, limit=True)
+    backward_query = create_story_query(item, uid_name, uid, reverse=True,
+                                        limit=True)
+
+    def _get_partial_story(query, resources_to_expand=None):
```

Comment:
We can remove the `=None` part now since we know this function will always be called with an appropriate value now.